### ISSUE TITLE
Searchbar tries to autofill in login data

### ISSUE THREAD

**Issue by therawk** [stephanrauh/ngx-extended-pdf-viewer#8]
Hi!

Great work on the library and thanks for sharing.

There is one issue which is weird - when I click on the search button the input field is pre-populated with a saved username from some login form. 

It is my email address and also I get 4  suggestions for other usernames. 

This is strange so I checked out the original viewer:

https://mozilla.github.io/pdf.js/web/viewer.html

I don't have the problem on there . Any idea? 🤔

**Comment by stephanrauh** [comment]
That's really weird. I have no idea what's happening. I don't see any autocompletions on my machine, nor do I see any reason why your username (of all things!) should be in the autocomplete list.

Be that as it may, I suppose it's a good idea to switch off the auto-completion list altogether. That's what I did just a few moments ago. Hope that helps!

**Comment by stephanrauh** [comment]
The bug fix is planned for version 0.9.8. Currently, this release is blocked by #6, but I intent to publish it soon.

**Comment by therawk** [comment]
Hi Stephan,
thanks for your effort! Unfortunately, the new version has the same issue. I guess it is related to chrome. In one of my forms I tried to use autocomplete="off" but Chrome seems to ignore it since - according to Google - "Users want to be able to save their passwords in forms". 

On Stackoverflow the recommendations were using either autocomplete="nope", autocomplete="new-password" or autocomplete="dfoijdfke" <- some random string.

I thought something might collide with my forms/app/anything. But here see the video of a fresh set-up application:

https://streamable.com/73r4h

As soon as the viewer loads the key appears next to the favourites button. I will fiddle around with the template this weekend and try to figure out what causes this. Not sure if you use FireFox or another browser where this problem doesn't occur (on my end, too).

Also since you said you don't have this issue, maybe take a look if the key appears in your chrome url bar as this might help you see if the problem still occurs.

I will get back to you as soon as I got some more free time but feel free to reply back I will read and try

Greets
Paul

**Comment by stephanrauh** [comment]
The problem might be the missing `name` attribute. Plus, adding a semantic `autofill="search-input-field"` sounds like a good idea.

**Comment by stephanrauh** [comment]
BTW, I'm using Chrome almost all of the time. But it's the Mac version, so there may be a difference.

**Comment by stephanrauh** [comment]
I've optimized the autofill settings in the 0.9.9 release. Hope it works now!

**Comment by stephanrauh** [comment]
I didn't manage to reproduce the bug. What do I have to do to see the error?

**Comment by stephanrauh** [comment]
And the login data stems from the password field of the PDF viewer?

Be that as it may, probably the best solution is to set `autocomplete="off"` or `autocomplete="new-password"` in your input field. Maybe your browser detects "my" password field and "your" input field and believes them to be username and password.

If https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete and https://stackoverflow.com/questions/2530/how-do-you-disable-browser-autocomplete-on-web-form-field-input-tags are correct, I also need to implement a bugfix. Four years ago, I followed an advice that specifically told me to set `autocomplete="some-unique-identifier"`, but it seems that advice targeted several Chrome versions that implemented the W3C standard in a non-standard way.

By the way, please apologize the sluggish answer: A nasty flu sent me to bed this week.

**Comment by vovka-s** [comment]
> And the login data stems from the password field of the PDF viewer?

No, the password field works correctly. Login data appears in my input after I close the PDF viewer...

> Be that as it may, probably the best solution is to set `autocomplete="off"` or `autocomplete="new-password"` in your input field. Maybe your browser detects "my" password field and "your" input field and believes them to be username and password.

I'll try to play with autocomplete settings.

> By the way, please apologize the sluggish answer: A nasty flu sent me to bed this week.

It's okay, I hope the flu subsided and now you are completely healthy

**Comment by danielehrhardt** [comment]
I have the same Issue. Still present with v14.0.4

**Comment by danielehrhardt** [comment]
> > And the login data stems from the password field of the PDF viewer?
> 
> No, the password field works correctly. Login data appears in my input after I close the PDF viewer...
> 
> > Be that as it may, probably the best solution is to set `autocomplete="off"` or `autocomplete="new-password"` in your input field. Maybe your browser detects "my" password field and "your" input field and believes them to be username and password.
> 
> I'll try to play with autocomplete settings.
> 
> > By the way, please apologize the sluggish answer: A nasty flu sent me to bed this week.
> 
> It's okay, I hope the flu subsided and now you are completely healthy

Nothing of these Solutions work for me.

**Comment by danielehrhardt** [comment]
> Sorry, my error is different from the one described in the issue
> 
> * I have an angular-ionic app
> * One of screens contains search input, some other elements and the button
> * pdf viewer is shown in separate modal window on click on the button
> * After closing the modal window the input field on the first screen become populated with login data
> 
> So the problem appears not on search bar in pdf viewer, but in other independent input field in the app

Same behavoir.

**Comment by stephanrauh** [comment]
Please update to version 14.0.5 and tell me if it solves the problem.

**Comment by stephanrauh** [comment]
Oh, and thanks for nudging me - I'd totally forgotten about this issue!

**Comment by danielehrhardt** [comment]
I see this Problem again in my Application.

**Comment by stephanrauh** [comment]
Are you using the current version? Was it gone in the meantime?

**Comment by HannesMittag** [comment]
I have the same issue, also using the current version

**Comment by stephanrauh** [comment]
Strange. 

@HannesMittag @danielehrhardt @mkvashennikov Would you mind to examine the HTML code of the `#findInput` field? My bugfix simply added the attribute `autocomplete="off"`. 

- Is it missing in your application? 
- If it's there, which browser are you using? 
- Or should I add a different attribute? I'm asking because every browser seems to interpret the attribute slightly different, and I can't even reproduce the bug on my machine.

Here's what it looks like on my machine:
```html
<input _ngcontent-qsa-c71="" autocomplete="off" id="findInput" title="Suchen" data-l10n-id="find_input" name="search-input-field" class="toolbarField" placeholder="Im Dokument suchen…" data-status="" aria-invalid="false">
```

<img width="370" alt="image" src="https://user-images.githubusercontent.com/3045767/188959233-9d999e15-2452-47eb-8029-952254069993.png">

**Comment by stephanrauh** [comment]
Version 15.0.0 sets the type of the password field to "hidden" except for the time the password field is shown. After that, the type is set to "hidden" again, and the value of the field is emptied. That should prevent browsers from autofilling login data in any of the input fields.

**Comment by danielehrhardt** [comment]
> That should prevent browsers from autofilling login data in any of the input fields.

What i do not understand is why this happens at all? What does the PDF View has to do with autocomplete?

**Comment by stephanrauh** [comment]
Nothing. But the browser can't know that. It simply sees there's a password field. It almost never shows - unless you're opening a password-protected document (see this demo: https://pdfviewer.net/extended-pdf-viewer/passwords).

Thing is, the password field is part of the DOM tree, and I can't easily change that. The browsers don't seem to mind that the password is hidden. They just see an ordinary input field with a low tab index and the next input field is the password field. That's enough to trick at least some browsers into believing the two fields are a combination of login field and password field.

Adding insult to injury, that's a not a standard. It's a proprietary addition of the browsers, so each browser manufacturer does it a bit differently, and tweaks the algorithms every once in a while.

**Comment by danielehrhardt** [comment]
> Nothing. But the browser can't know that. It simply sees there's a password field. It almost never shows - unless you're opening a password-protected document (see this demo: https://pdfviewer.net/extended-pdf-viewer/passwords).
> 
> Thing is, the password field is part of the DOM tree, and I can't easily change that. The browsers don't seem to mind that the password is hidden. They just see an ordinary input field with a low tab index and the next input field is the password field. That's enough to trick at least some browsers into believing the two fields are a combination of login field and password field.
> 
> Adding insult to injury, that's a not a standard. It's a proprietary addition of the browsers, so each browser manufacturer does it a bit differently, and tweaks the algorithms every once in a while.

Ah I understand! :D

**Comment by sureshJaggarapu** [comment]
I am getting the page error while generating pdf, find the details in below screenshot.
![image](https://github.com/user-attachments/assets/6bd5ef03-dafd-43fd-ad66-67769d057007)

**Comment by stephanrauh** [comment]
@sureshJaggarapu I've opend a ticket. IMHO that's a very strange but because the field `#password` is present in the DOM.

BTW, I'm afraid I can't support old versions of ngx-extended-pdf-viewer. I'd love to do so, but I don't have enough time to spare, so I focus on the latest version of the library. That's why I've scheduled your issue to version 21 despite knowing you're using an older version.

**Comment by sureshJaggarapu** [comment]
Thank you so much @stephanrauh, I can understand, it is okay for me even though bug will resolve in latest version. we have a plan to upgrade  application version in next month.

**Comment by stephanrauh** [comment]
I think it's basically a Chrome bug, but even so, I believe to have found a solution. It's landed with version 21.0.0-alpha.12.
